### PR TITLE
add field icon in DnD tree

### PR DESCRIPTION
**Pull request by 3nids**
image

Really useful when you don't want to distinguish joined fields, or more generally to show the field type.

Can this be seen as bugfix?

### CODE REVIEW

**Comment by nyalldawson** [comment]
Yes -- great idea!!

(I wonder if we could get icons from qt designer for the widget side too...)

File: src/core/qgsfields.h [hunk 1]
```diff
@@ -321,9 +321,11 @@ class CORE_EXPORT  QgsFields
 
     /**
      * Returns an icon corresponding to a field index, based on the field's type and source
+     * \param fieldIdx the field index
+     * \param considerOrigin, if True the icon will the origin of the field
```
Comment (nyalldawson) [inline src/core/qgsfields.h:325]:
```suggestion
     * \param considerOrigin if TRUE the icon will the origin of the field
```

**Comment by elpaso** [comment]
Cool! Thank you.

**Review by nyalldawson — APPROVED** [review]

File: src/gui/vector/qgsattributesformproperties.cpp [hunk 2]
```diff
@@ -97,12 +97,12 @@ void QgsAttributesFormProperties::initAvailableWidgetsTree()
   for ( int i = 0; i < fields.size(); ++i )
   {
     const QgsField field = fields.at( i );
-    DnDTreeItemData itemData = DnDTreeItemData( DnDTreeItemData::Field, field.name(), field.name() );
+    DnDTreeItemData itemData = DnDTreeItemData( DnDTreeItemData::Field, field.name(), field.name(), QColor() );
```
Comment (3nids) [inline src/gui/vector/qgsattributesformproperties.cpp:100]:
```suggestion
    DnDTreeItemData itemData = DnDTreeItemData( DnDTreeItemData::Field, field.name(), field.name() );
```

**Pull request merged**